### PR TITLE
Sync git submodules with Guix

### PR DESCRIPTION
**Pull request by aadcg**
# Description

The sync is key, as to ensure that CL dependencies match between the Guix dev environment and the manual installation followed by users. 

# Discussion

Can only be merged _after_ the patches below are accepted:

- https://debbugs.gnu.org/cgi/bugreport.cgi?bug=64701
- https://debbugs.gnu.org/cgi/bugreport.cgi?bug=64702
- https://debbugs.gnu.org/cgi/bugreport.cgi?bug=64703
- https://debbugs.gnu.org/cgi/bugreport.cgi?bug=64704

**Update:** All have been accepted.

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [x] Git hygiene:
  - I have pulled from master before submitting this PR
  - There are no merge conflicts.
- [x] I've added the new dependencies as:
  - ASDF dependencies,
  - Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - and Guix dependencies.
- [x] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [x] I have performed a self-review of my own code.
- [ ] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [x] Documentation:
  - All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - I have updated the existing documentation to match my changes.
  - I have commented my code in hard-to-understand areas.
  - I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
    - Changelog update should be a separate commit.
  - I have added a `migration.lisp` entry for all compatibility-breaking changes.
  - (If this changes something about the features showcased on Nyxt website) I have these changes described in the new/existing article at Nyxt website or will notify one of maintainters to do so.
- [x] Compilation and tests:
  - My changes generate no new warnings.
  - I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - I ran the tests locally (`(asdf:test-system :nyxt)` and `(asdf:test-system :nyxt/gi-gtk)`) and they pass.

### CODE REVIEW

**Comment by jmercouris** [comment]
I will recompile.

**Comment by jmercouris** [comment]
Instantly resulted in failure:

```
Submodule path '_build/bordeaux-threads': checked out '076fe2380abbc59b06e495dc7a35aea8eb26ba3b'
Submodule path '_build/calispel': checked out 'e9f2f9c1af97f4d7bb4c8ac25fb2a8f3e8fada7a'
Submodule path '_build/cffi': checked out '33970351e71bb5f12ba56fc40270089e948ae112'
fatal: remote error: upload-pack: not our ref 82a17d39c78d91f6ea63a03aca8f9aa6069a5e11
fatal: Fetched in submodule path '_build/chipz', but it did not contain 82a17d39c78d91f6ea63a03aca8f9aa6069a5e11. Direct fetching of that commit failed.
; CL_SOURCE_REGISTRY: "/home/jmercouris/Work/Atlas/Nyxt/libraries/nasdf/:/home/jmercouris/Work/Atlas/Nyxt/_build//:/home/jmercouris/Work/Atlas/Nyxt/:/home/jmercouris/Work/Atlas/Nyxt/libraries/nasdf/:"
; NASDF 0.1.6: /home/jmercouris/Work/Atlas/Nyxt/libraries/nasdf/nasdf.asd
Unhandled ASDF/FIND-COMPONENT:MISSING-DEPENDENCY in thread #<SB-THREAD:THREAD tid=58675 "main thread" RUNNING
                                                              {1001088003}>:
  Component ASDF/USER::CL-GOBJECT-INTROSPECTION not found, required by
  #<NASDF-SYSTEM "nyxt/gi-gtk">

Backtrace for: #<SB-THREAD:THREAD tid=58675 "main thread" RUNNING {1001088003}>
0: (SB-DEBUG::DEBUGGER-DISABLED-HOOK Component ASDF/USER::CL-GOBJECT-INTROSPECTION not found, required by #<NASDF-SYSTEM "nyxt/gi-gtk"> #<unused argument> :QUIT T)
1: (SB-DEBUG::RUN-HOOK *INVOKE-DEBUGGER-HOOK* Component ASDF/USER::CL-GOBJECT-INTROSPECTION not found, required by #<NASDF-SYSTEM "nyxt/gi-gtk">)
2: (INVOKE-DEBUGGER Component ASDF/USER::CL-GOBJECT-INTROSPECTION not found, required by #<NASDF-SYSTEM "nyxt/gi-gtk">)
3: (ERROR ASDF/FIND-COMPONENT:MISSING-DEPENDENCY :REQUIRED-BY #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk"> :REQUIRES ASDF/USER::CL-GOBJECT-INTROSPECTION)
4: (ASDF/FIND-COMPONENT:RESOLVE-DEPENDENCY-NAME #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk"> ASDF/USER::CL-GOBJECT-INTROSPECTION NIL)
5: (ASDF/PLAN:MAP-DIRECT-DEPENDENCIES #<ASDF/LISP-ACTION:PREPARE-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk"> #<FUNCTION (LAMBDA (ASDF/PLAN::O ASDF/PLAN::C) :IN ASDF/PLAN:TRAVERSE-ACTION) {100A3C940B}>)
6: ((LAMBDA NIL :IN ASDF/PLAN:TRAVERSE-ACTION))
7: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
8: (ASDF/PLAN:TRAVERSE-ACTION #<ASDF/PLAN:SEQUENTIAL-PLAN {100A39ADE3}> #<ASDF/LISP-ACTION:PREPARE-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk"> T)
9: (ASDF/PLAN:MAP-DIRECT-DEPENDENCIES #<ASDF/LISP-ACTION:PREPARE-OP > #<ASDF/LISP-ACTION:CL-SOURCE-FILE "nyxt/gi-gtk" "renderer/gi-gtk"> #<FUNCTION (LAMBDA (ASDF/PLAN::O ASDF/PLAN::C) :IN ASDF/PLAN:TRAVERSE-ACTION) {100A3BB56B}>)
10: ((LAMBDA NIL :IN ASDF/PLAN:TRAVERSE-ACTION))
11: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
12: (ASDF/PLAN:TRAVERSE-ACTION #<ASDF/PLAN:SEQUENTIAL-PLAN {100A39ADE3}> #<ASDF/LISP-ACTION:PREPARE-OP > #<ASDF/LISP-ACTION:CL-SOURCE-FILE "nyxt/gi-gtk" "renderer/gi-gtk"> T)
13: (ASDF/PLAN:MAP-DIRECT-DEPENDENCIES #<ASDF/LISP-ACTION:LOAD-OP > #<ASDF/LISP-ACTION:CL-SOURCE-FILE "nyxt/gi-gtk" "renderer/gi-gtk"> #<FUNCTION (LAMBDA (ASDF/PLAN::O ASDF/PLAN::C) :IN ASDF/PLAN:TRAVERSE-ACTION) {100A3BB3EB}>)
14: ((LAMBDA NIL :IN ASDF/PLAN:TRAVERSE-ACTION))
15: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
16: (ASDF/PLAN:TRAVERSE-ACTION #<ASDF/PLAN:SEQUENTIAL-PLAN {100A39ADE3}> #<ASDF/LISP-ACTION:LOAD-OP > #<ASDF/LISP-ACTION:CL-SOURCE-FILE "nyxt/gi-gtk" "renderer/gi-gtk"> T)
17: (ASDF/PLAN:MAP-DIRECT-DEPENDENCIES #<ASDF/LISP-ACTION:LOAD-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk"> #<FUNCTION (LAMBDA (ASDF/PLAN::O ASDF/PLAN::C) :IN ASDF/PLAN:TRAVERSE-ACTION) {100A3BB24B}>)
18: ((LAMBDA NIL :IN ASDF/PLAN:TRAVERSE-ACTION))
19: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
20: (ASDF/PLAN:TRAVERSE-ACTION #<ASDF/PLAN:SEQUENTIAL-PLAN {100A39ADE3}> #<ASDF/LISP-ACTION:LOAD-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk"> T)
21: (ASDF/PLAN:MAP-DIRECT-DEPENDENCIES #<ASDF/LISP-ACTION:PREPARE-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk-application"> #<FUNCTION (LAMBDA (ASDF/PLAN::O ASDF/PLAN::C) :IN ASDF/PLAN:TRAVERSE-ACTION) {100A3ACACB}>)
22: ((LAMBDA NIL :IN ASDF/PLAN:TRAVERSE-ACTION))
23: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
24: (ASDF/PLAN:TRAVERSE-ACTION #<ASDF/PLAN:SEQUENTIAL-PLAN {100A39ADE3}> #<ASDF/LISP-ACTION:PREPARE-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk-application"> T)
25: (ASDF/PLAN:MAP-DIRECT-DEPENDENCIES #<ASDF/LISP-ACTION:LOAD-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk-application"> #<FUNCTION (LAMBDA (ASDF/PLAN::O ASDF/PLAN::C) :IN ASDF/PLAN:TRAVERSE-ACTION) {100A3AC92B}>)
26: ((LAMBDA NIL :IN ASDF/PLAN:TRAVERSE-ACTION))
27: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
28: (ASDF/PLAN:TRAVERSE-ACTION #<ASDF/PLAN:SEQUENTIAL-PLAN {100A39ADE3}> #<ASDF/LISP-ACTION:LOAD-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk-application"> T)
29: ((LAMBDA NIL :IN ASDF/PLAN:MAKE-PLAN))
30: ((:METHOD ASDF/OPERATE:OPERATE (ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) #<ASDF/LISP-ACTION:LOAD-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk-application"> :PLAN-CLASS NIL :PLAN-OPTIONS NIL) [fast-method]
31: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:LOAD-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk-application">)
32: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
33: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) #<ASDF/LISP-ACTION:LOAD-OP > #<NASDF:NASDF-SYSTEM "nyxt/gi-gtk-application">) [fast-method]
34: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> ASDF/LISP-ACTION:LOAD-OP :NYXT/GI-GTK-APPLICATION)
35: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
36: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:LOAD-OP :NYXT/GI-GTK-APPLICATION) [fast-method]
37: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {100A35AE8B}> :OVERRIDE T :KEY NIL :OVERRIDE-CACHE T :OVERRIDE-FORCING NIL)
38: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
39: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<FUNCTION (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {100A3274CB}> :OVERRIDE NIL :KEY NIL :OVERRIDE-CACHE NIL :OVERRIDE-FORCING NIL)
40: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:LOAD-OP :NYXT/GI-GTK-APPLICATION) [fast-method]
41: (ASDF/OPERATE:LOAD-SYSTEM :NYXT/GI-GTK-APPLICATION)
42: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ASDF/OPERATE:LOAD-SYSTEM :NYXT/GI-GTK-APPLICATION) #<NULL-LEXENV>)
43: (EVAL (ASDF/OPERATE:LOAD-SYSTEM :NYXT/GI-GTK-APPLICATION))
44: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(require \"asdf\")") (:EVAL . #<(SIMPLE-ARRAY CHARACTER (336)) (when (string= "true" "true") (setf asdf:*default-source-registries* (list (quote asdf/source-registry:environment-source-registry))) (asdf:clear-configuration) (asdf:load-asd "/home/jmercouris/Work/A... {10010984FF}>) (:EVAL . "(asdf:load-asd \"/home/jmercouris/Work/Atlas/Nyxt//libraries/nasdf/nasdf.asd\")") (:EVAL . "(asdf:load-asd \"/home/jmercouris/Work/Atlas/Nyxt//nyxt.asd\")") (:EVAL . "(format t \"~&; NASDF ~a: ~a~&\" (asdf:component-version (asdf:find-system :nasdf)) (nth-value 2 (asdf:locate-system :nasdf)))") (:EVAL . "(when (find-package :ql) (funcall (read-from-string \"ql:quickload\") :cffi))") (:EVAL . #<(SIMPLE-ARRAY CHARACTER (235)) (when (and (find-package :cffi) (uiop:getenv "GUIX_ENVIRONMENT")) (pushnew (pathname (format nil "~a/lib/" (uiop:getenv "GUIX_ENVIRONMENT"))) (symbol-value (read-from-string "cffi:*foreign-library-dir... {10010990CF}>) (:EVAL . "(asdf:load-system :nyxt/gi-gtk-application)") (:EVAL . "(asdf:make :nyxt/gi-gtk-application)") (:EVAL . "(uiop:quit)") (:QUIT)))
45: (SB-IMPL::TOPLEVEL-INIT)
46: ((FLET SB-UNIX::BODY :IN SB-IMPL::START-LISP))
47: ((FLET "WITHOUT-INTERRUPTS-BODY-3" :IN SB-IMPL::START-LISP))
48: (SB-IMPL::%START-LISP)

unhandled condition in --disable-debugger mode, quitting

Compilation failed, see the above stacktrace.

make: *** [makefile:54: nyxt] Error 1
```

**Comment by jmercouris** [comment]
My _build folder was already empty.

**Comment by jmercouris** [comment]
Same thing:

```
fatal: remote error: upload-pack: not our ref 82a17d39c78d91f6ea63a03aca8f9aa6069a5e11
fatal: Fetched in submodule path '_build/chipz', but it did not contain 82a17d39c78d91f6ea63a03aca8f9aa6069a5e11. Direct fetching of that commit failed.
```

File: .gitmodules [hunk 1]
```diff
@@ -601,3 +589,14 @@
 	path = _build/cl-colors2
 	url = https://notabug.org/cage/cl-colors2
 	shallow = false # Notabug does not advertize commits.
+[submodule "_build/chipz"]
+	path = _build/chipz
+	url = https://github.com/sharplispers/chipz
```
Comment (Ambrevar) [inline .gitmodules:594]: `shallow = true` is missing.

**Review by Ambrevar — COMMENTED** [review]
How did you update the submodules?  Are they in sync with the packages from a specific Guix commit?  If so, which one?
This makes me realize that we should record this Guix commit somewhere...

**Comment by Ambrevar** [comment]
Alrighty.

One last question you didn't answer: how did you perform the sync?

There is `build-scripts/sync-submodules-with-guix.scm`, but it's a bit incomplete, it didn't work out of the  box last time I tried.

**Comment by jmercouris** [comment]
It worked in a fresh checkout.

**Review by jmercouris — APPROVED** [review]

**Comment by Ambrevar** [comment]
> Since the submodule are shallowed cloned (the reasons being clear), sometimes I had to locally unshallow a submodule so that the commit I wanted to point to would be found.

Yes, I ran into the same issue.  It would be nice to figure out a smooth way to update a shallow module to a specific commit.

Out of curiosity, how did you "unshallow"?

**Comment by jmercouris** [comment]
Yes, let's merge!

**Comment by Ambrevar** [comment]
> > 
> 
> `git fetch --unshallow`.

Duh, I didn't know this command!  Excellent!  Thanks for sharing!

And thanks for the the work on this pull request, it's been long due!

**Pull request merged**